### PR TITLE
Convert between LocalDateTime and ZonedDateTime - BAEL-5879

### PR DESCRIPTION
**Pull request by olu-damilare**

### CODE REVIEW

File: core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    }
+
+    @Test
+    void whenConvertLocalDateTimeToZonedDateTimeWithOfMethod_shouldConvert(){
+        LocalDateTime localDateTime = LocalDateTime.of(2022, 11, 5, 7, 30, 22);
+        ZonedDateTime zonedDateTime = ZonedDateTime.of(localDateTime, ZoneId.of("Africa/Accra")).withZoneSameInstant(ZoneId.of("Africa/Lagos"));
+
+        assertEquals("2022-11-05T07:30:22", localDateTime.toString());
+        assertEquals("2022-11-05T08:30:22+01:00[Africa/Lagos]", zonedDateTime.toString()); assertEquals(localDateTime.getYear(), zonedDateTime.getYear());
```
Comment (KevinGilmore) [inline core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java:42]: You need to put the second statement on its own line

File: core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        assertEquals(localDateTime.getMinute(), zonedDateTime.getMinute());
+        assertEquals(localDateTime.getSecond(), zonedDateTime.getSecond());
+    }
+
+    @Test
+    void whenConvertLocalDateTimeToZonedDateTimeWithAtZoneAndwithZoneSameInstantMethods_shouldConvert(){
+        LocalDateTime localDateTime = LocalDateTime.of(2022, 1, 1, 0, 30, 22);
+        ZonedDateTime zonedDateTime = localDateTime.atZone(ZoneId.of("Africa/Lagos")).withZoneSameInstant(ZoneId.of("Canada/Atlantic"));
+
+        assertEquals("2022-01-01T00:30:22", localDateTime.toString());
```
Comment (KevinGilmore) [inline core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java:30]: We probably don't need this assertion and others like it below that involve asserting something on the given conditions, which we already know.

File: core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        assertEquals("2021-12-31T19:30:22-04:00[Canada/Atlantic]", zonedDateTime.toString());
+        assertEquals("-04:00", zonedDateTime.getOffset().toString());
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java:33]: Let's get rid of the blank lines before closing braces

File: core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    @Test
+    void whenConvertZonedDateTimeToLocalDateTimeWithToLocalDateTimeMethod_shouldConvert(){
+        ZonedDateTime zonedDateTime = ZonedDateTime.of(2011, 2, 12, 6, 14, 1, 58086000, ZoneId.of("Asia/Tokyo"));
+        LocalDateTime localDateTime = zonedDateTime.toLocalDateTime();
+
+        assertEquals("2011-02-12T06:14:01.058086", localDateTime.toString());
+        assertEquals("2011-02-12T06:14:01.058086+09:00[Asia/Tokyo]", zonedDateTime.toString());
+    }
+
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java:101]: Let's remove these blank lines

File: core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java (hunk 2 above)
Comment (olu-damilare) [reply core-java-modules/core-java-datetime-conversion/src/test/java/com/baeldung/localdatetimetozoneddatetime/LocalDateTimeToZonedDateTimeUnitTest.java:30]: @KevinGilmore I was advised to include this by the editor. I reckon that it improves the reader's understanding, so we can just leave it as is.

**Review by saajn — APPROVED** [review]

**Pull request merged**